src/devices/ExplorerHat/Lights.cs

```diff
@@ -104,24 +106,22 @@ public void Off()
 
         #region IDisposable Support
 
-        private bool _disposedValue = false; // Para detectar llamadas redundantes
+        private bool _shouldDispose;
 
         /// <summary>
         /// Disposes the <see cref="Lights"/> instance
         /// </summary>
         protected virtual void Dispose(bool disposing)
         {
-            if (!_disposedValue)
+            if (disposing)
```

Comment:
I think you shouldn't change this condition from my comment above. This was really just meant to catch the case of double call to Dispose()